### PR TITLE
Fix CartesianProductTest `factory` not working with fully-qualified names

### PR DESCRIPTION
**Pull request by Michael1993**
Proposed commit message:

```
Fix `CartesianProductTest` factory method (#412 / #418)

This commit fixes the `CartesianProductTest#factory` method not
working with fully-qualified method names, e.g.:
`org.sample.Class#factoryMethod`

Closes: #412
PR: #418
```

---
**PR checklist**

The following checklist shall help the PR's author, the reviewers and maintainers to ensure the quality of this project.
It is based on our contributors guidelines, especially the ["writing code" section](https://github.com/junit-pioneer/junit-pioneer/blob/master/CONTRIBUTING.md#writing-code).
It shall help to check for completion of the listed points.
If a point does not apply to the given PR's changes, the corresponding entry can be simply marked as done. 

Documentation (general)
* [ ] There is documentation (Javadoc and site documentation; added or updated)
* [ ] There is implementation information to describe _why_ a non-obvious source code / solution got implemented
* [ ] Site documentation has its own `.adoc` file in the `docs` folder, e.g. `docs/report-entries.adoc`
* [ ] Only one sentence per line (especially in `.adoc` files)
* [ ] Javadoc uses formal style, while sites documentation may use informal style

Code
* [X] Code adheres to code style, naming conventions etc.
* [X] Successful tests cover all changes
* [X] There are checks which validate correct / false usage / configuration of a functionality and there are tests to verify those checks
* [X] Tests use [AssertJ](https://joel-costigliola.github.io/assertj/) or our own [PioneerAssert](https://github.com/junit-pioneer/junit-pioneer/blob/master/CONTRIBUTING.md#assertions) (which are based on AssertJ)

Contributing
* [X] A prepared commit message exists

---

I hereby agree to the terms of the [JUnit Pioneer Contributor License Agreement](https://github.com/junit-pioneer/junit-pioneer/blob/master/CONTRIBUTING.md#junit-pioneer-contributor-license-agreement).

### CODE REVIEW

**Comment by Michael1993** [comment]
TODO:

* [ ] Update documentation.

File: src/main/java/org/junitpioneer/jupiter/CartesianProductTestExtension.java [hunk 1]
```diff
@@ -169,6 +170,26 @@ private Method findSetsFactory(Method testMethod, String factoryMethodName) {
...
+		if (factoryMethodName.contains("#")) {
+			String className = factoryMethodName.substring(0, factoryMethodName.indexOf('#'));
+			return ReflectionSupport
+					.tryToLoadClass(className)
+					.getOrThrow(ex -> new ExtensionConfigurationException(
+						format("Class %s not found, referenced in method %s", className, testMethod.getName()), ex));
+
+		}
+		return testMethod.getDeclaringClass();
+	}
```
Comment (nipafx) [inline src/main/java/org/junitpioneer/jupiter/CartesianProductTestExtension.java:191]: Is there no tooling we can use instead of string manipulation?

**Review by nipafx — COMMENTED** [review]
I merged #425, which I guess caused the merge conflict.

File: src/main/java/org/junitpioneer/jupiter/CartesianProductTestExtension.java (hunk 1 above)
Comment (Michael1993) [reply src/main/java/org/junitpioneer/jupiter/CartesianProductTestExtension.java:191]: Do you have a suggestion? I just copy-pasted from JUnit's `@MethodSource`.

Comment (nipafx) [reply src/main/java/org/junitpioneer/jupiter/CartesianProductTestExtension.java:191]: Nope, it was an honest (open) question. String manipulation just feels so yucky.

Maybe there's an OSS project hiding here - either already existing or desperately needed. 😉

**Review by Bukama — APPROVED** [review]

**Pull request merged**